osquery/tables/system/darwin/xprotect.cpp

```diff
@@ -144,43 +145,50 @@ QueryData genXProtectReports(QueryContext& context) {
   return results;
 }
 
-QueryData genXProtectEntries(QueryContext& context) {
-  QueryData results;
+void genXProtectEntriesFromPath(const std::string& xprotect_dir,
+                                QueryData& results) {
   pt::ptree tree;
 
-  auto xprotect_path = fs::path(kXProtectPath) / "XProtect.plist";
+  auto xprotect_path = fs::path(xprotect_dir) / "XProtect.plist";
   if (!osquery::pathExists(xprotect_path).ok()) {
     VLOG(1) << "XProtect.plist is missing";
```

Comment:
It seems a little wrong to always log this message at least once, even if a plist _is_ found elsewhere. We should probably only print it out if the plist or meta.plist isn't found anywhere. (note to self)

Reply:
Right, I would bring this file path exists check out of this inner function and into `genXProtectMeta`. Then keep a flag `plist_exists` and set it to true if any of the files exist. Then at the end of the method if no files exist, emit this log.

Reply:
Same with `genXProtectEntries`.

I see some overlap between these methods, maybe do a quick check to see if we can refactor and reduce the amount of code? I don't have explicit ideas on what to do, my ask if that while you're improving this, could you do a check?

Reply:
Yea I can take a look for sure